awx/ui_next/src/components/LaunchPrompt/steps/useSurveyStep.jsx

```diff
@@ -1,33 +1,34 @@
-import React, { useState, useEffect, useCallback } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import { t } from '@lingui/macro';
+import { useFormikContext } from 'formik';
 import useRequest from '../../../util/useRequest';
 import { JobTemplatesAPI, WorkflowJobTemplatesAPI } from '../../../api';
 import SurveyStep from './SurveyStep';
 import StepName from './StepName';
 
 const STEP_ID = 'survey';
 
-export default function useSurveyStep(config, resource, visitedSteps, i18n) {
-  const [stepErrors, setStepErrors] = useState({});
-
+export default function useSurveyStep(config, visitedSteps, i18n) {
+  const { values } = useFormikContext();
   const { result: survey, request: fetchSurvey, isLoading, error } = useRequest(
     useCallback(async () => {
       if (!config.survey_enabled) {
         return {};
       }
-      const { data } =
-        resource.type === 'workflow_job_template'
-          ? await WorkflowJobTemplatesAPI.readSurvey(resource.id)
-          : await JobTemplatesAPI.readSurvey(resource.id);
+      const { data } = config?.workflow_job_template_data
+        ? await WorkflowJobTemplatesAPI.readSurvey(
+            config?.workflow_job_template_data?.id
+          )
+        : await JobTemplatesAPI.readSurvey(config?.job_template_data?.id);
       return data;
-    }, [config.survey_enabled, resource])
+    }, [config])
   );
 
   useEffect(() => {
     fetchSurvey();
   }, [fetchSurvey]);
 
-  const validate = values => {
+  const validate = () => {
```

Comment:
do you still need this validate function? It doesn't look like `useLaunchSteps` does anything with it anymore